.github/workflows/main.yml

```diff
@@ -32,17 +32,46 @@ jobs:
         with:
           submodules: 'true'
 
+      # Special handling to test against Java 8:
+      # antlr version 4.10 requires Java 11 to run the antlr tool, but tests still need to run
+      # against Java 8 - that's being done here:
+      - name: Set up Java 8
+        if: ${{ matrix.java-version == '11' }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 8
+      - name: Configure env with Java 8 home
+        if: ${{ matrix.java-version == '11' }}
+        run:
+          echo "JAVA_8_HOME=$JAVA_HOME" >> $GITHUB_ENV
+      - name: Enable Java 8 tests
+        if: ${{ matrix.java-version == '11' }}
+        run:
+          echo "ADDITIONAL_GRADLE_OPTS=-PalsoTestAgainstJava8" >> $GITHUB_ENV
+      - name: Configure Java 8 home for Gradle
+        if: ${{ matrix.java-version == '11' }}
+        run: |
+          mkdir -p $HOME/.gradle
+          echo "org.gradle.java.installations.fromEnv=JAVA_8_HOME" >> $HOME/.gradle/gradle.properties
+
       - name: Set up JDK ${{ matrix.java-version }}
         uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java-version }}
 
+      - name: Spotless Check
+        uses: gradle/gradle-build-action@v2
+        with:
+          # Spotless must run in a different invocation, because
+          # it has some weird Gradle configuration/variant issue
+          arguments: spotlessCheck
+
       - name: Build with Gradle
-        # Java 8 and 16+ don't run spotless, because google-java-format requires Java 11+ (but doesn't run w/ Java 16)
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: --rerun-tasks --info assemble check publishToMavenLocal -x jmh
+          arguments: --rerun-tasks assemble ${{ env.ADDITIONAL_GRADLE_OPTS }} check publishToMavenLocal -x jmh -x spotlessCheck
```

Comment:
question: do the opts need to be inbetween task names? or could they come after `--rerun-tasks` instead?

Reply:
The order doesn't matter in this case.